### DISCUSSION TITLE
[QUESTION] - Is it possible to get information for function arguments?

### DISCUSSION THREAD

**Discussion by sujit-baniya** [antonmedv/expr#469] [General]
I've a requirement to get information about a function and it's arguments.

For e.g. `age(customer.dob, 'Male')`

Here I wanted to identify above something like

```json
{
    "func_o": "age",
    "arguments": [
    {"argument": "customer.dob","type": "variable"},
    {"argument": "'Male'","type": "constant"}
]
}

```

**Comment by antonmedv** [comment]
Yes. It is totally possible. Just use the visitor https://expr.medv.io/docs/Visitor-and-Patch

**Comment by sujit-baniya** [comment]
@antonmedv Thanks. I will check it. Is it fine to identify `a.b` (nested variable) as constant?

**Comment by antonmedv** [comment]
Not sure what you asking about.

**Comment by sujit-baniya** [comment]
I meant, When I try to Parse `a.b` or any variables (identifiers), it is parsed as `Constants` instead of `Variables`

```go
	v, err := expr.Parse("a.b")
	if err != nil {
		panic(err)
	}
	fmt.Println(v.Variables, v.Constants)
```

IMO, 

* `'a.b'` - as constant (currently is fine)
* `a.b` - as variable (ISSUE)
* `true` - Is neither variable nor constant (ISSUE) it should have been a constant

**Comment by antonmedv** [comment]
`*vm.Program` "constants" is not what you think. And "variables" is not vars. It is internal VM structures. I guess, I need to rename those.
